### PR TITLE
Fix #9043 - Ensure QR code scanner works

### LINKED ISSUE #9043: QR code scanning function not working
**Describe the bug**
MetaMask extension 8.x qr code scan for recipient's address is not functioning on any browser (firefox, brave, or chrome).

**To Reproduce (REQUIRED)**
Click Send, select the qr scan button.

image

**Browser details (please complete the following information):**
 - OS: Mac OS, did not test on windows or linux
 - Hardware Wallet [e.g. Trezor Firmware version 1.8.3, Ledger Nano S Firmware version 1.6.0]
 - Browser [firefox, brave, and chrome]
 - MetaMask Version [8.x]

### PR DESCRIPTION
**Pull request by darkwing**
This pull requests ensures the QR code reader functions properly in Chrome and Firefox, with and without camera permissions at the beginning of the process.

### CODE REVIEW

**Comment by darkwing** [comment]
I'm not in love with this solution yet but it works in Firefox and Chrome, both with the camera starting off as "blocked" and then enabled.

The teardown, unfortunately, is required for Firefox, otherwise it attempts to create two video streams.  I'll continue working toward a better solution.

**Comment by darkwing** [comment]
To test:

1.  Open Firefox, click "Send", then click the scanner icon --- video should load.
2.  Revoke camera permissions, repeat step 1, ensure video loads

3.  Open Chrome, click "Send", then click the scanner icon --- video should load.
4.  Revoke camera permissions, repeat step 3, give camera permissions, wait a moment and camera should load.

File: ui/app/components/app/modals/qr-scanner/qr-scanner.component.js [hunk 1]
```diff
@@ -104,15 +104,29 @@ export default class QrScanner extends Component {
...
+    // the user the camera permission request.  We must then call it again
+    // once we receive permission so that the video displays.
+    // It's important to prevent this codeReader from being created twice;
+    // Firefox otherwise starts 2 video streams, one of which cannot be stopped
+    if (!this.codeReader) {
+      this.codeReader = new BrowserQRCodeReader()
+    }
     try {
       await this.codeReader.getVideoInputDevices()
+      this.checkPermissions()
```
Comment (darkwing) [inline ui/app/components/app/modals/qr-scanner/qr-scanner.component.js:129]: Kicking off this async all starts the component listening for the connection if permissions have not yet been given.

**Comment by darkwing** [comment]
This is good to go!

**Comment by brad-decker** [comment]
Confirmed that this does resolve the issue with the camera on both firefox and chrome. I first recreated the issue on the develop branch, then switched to this PR and tried again with success. Well done. I'm looking through the code changes now

**Comment by brad-decker** [comment]
@darkwing in an above comment you mentioned you weren't in love with this solution -- does that reflect this current change? If so, what would you improve upon?

**Comment by darkwing** [comment]
I fell more in love with it after a few changes :), so I no longer feel that way.  I do dislike how you have to call the camera to get the prompt, but we can live with it as long as it works for our users!

**Review by brad-decker — APPROVED** [review]
This all looks good to me, makes sense that we need to do proper teardown and what not to avoid multiple video channels. Thank you for your detailed comments and explanation for your changes.

**Pull request merged**